### PR TITLE
feat: edit the field by clicking on the drag icon

### PR DESCRIPTION
**Pull request by ekakshjanweja**
### Feature Preview

---

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [x] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

### CODE REVIEW

**Comment by richardshiue** [comment]
Hi there @ekakshjanweja, great progress! I have some feedback:

- There's no need to update the pubspec.lock, and I also think importing dart:io isn't necessary here (and renaming log as well)
- The tooltip text needs to be through i18n

What do you think about the use of blockactionbutton here @LucasXu0 ?

**Comment by Xazin** [comment]
Seems something went wrong when rebasing/merging main.

Your PR contains changes of earlier commits, and there is still an unresolved conflict.

If you don't mind I can resolve it for you, let me know.

**Comment by Xazin** [comment]
@ekakshjanweja I've fixed it, but it is now "my" commit, do this to take ownership:

0. Make sure you don't have any changes on your current branch
1. Check out the current state of the branch:
```
git fetch
git reset --hard origin/edit-field-by-drag-icon
```
2. Make sure you don't have any non-committed changes, and then
```
git reset --soft HEAD~1
git commit -m "feat: edit the field by clicking on the drag icon"
git push --force
```

---

The above will alter the commit history, thus needing you to push with force. It's the same I did above.

**Comment by LucasXu0** [comment]
@ekakshjanweja Thank you for your contribution. Please sign the CLA and attach a screenshot to demonstrate this feature.

Here are the steps for updating the translation.

1. add your translations to `frontend/resources/translations/en.json`
2. ```cd frontend && sh scripts/code_generation/language_files/generate_language_files.sh```
3. replace the hardcode string with translation key.

**Review by richardshiue — CHANGES_REQUESTED** [review]
The block action button is supposed to be hidden when not hovering over that row in the property list.

Screenshot 2023-10-12 at 10 41 22 AM

**Comment by ekakshjanweja** [comment]
> The block action button is supposed to be hidden when not hovering over that row in the property list.
> 
> Screenshot 2023-10-12 at 10 41 22 AM

Okay I will look into it.

**Comment by ekakshjanweja** [comment]
@richardshiue I addressed the requested changes.

File: frontend/appflowy_flutter/lib/plugins/database_view/widgets/row/row_property.dart [hunk 1]
```diff
@@ -140,7 +143,28 @@ class _PropertyCellState extends State<_PropertyCell> {
       child: SizedBox(
         width: 16,
         height: 30,
-        child: _isFieldHover ? const FlowySvg(FlowySvgs.drag_element_s) : null,
+        child: AppFlowyPopover(
+          controller: _popoverController,
```
Comment (richardshiue) [inline frontend/appflowy_flutter/lib/plugins/database_view/widgets/row/row_property.dart:147]: Don't use the same _popoverController as below. This makes the popover appear in the wrong place.

File: frontend/appflowy_flutter/lib/plugins/database_view/widgets/row/row_property.dart [hunk 2]
```diff
@@ -140,7 +143,28 @@ class _PropertyCellState extends State<_PropertyCell> {
...
+          constraints: BoxConstraints.loose(const Size(240, 600)),
+          margin: EdgeInsets.zero,
+          triggerActions: PopoverTriggerFlags.none,
+          direction: PopoverDirection.bottomWithLeftAligned,
+          popupBuilder: (popoverContext) => buildFieldEditor(),
+          child: SizedBox(
+            width: 160,
```
Comment (richardshiue) [inline frontend/appflowy_flutter/lib/plugins/database_view/widgets/row/row_property.dart:154]: We probably don't need a 160px wide sized box here

File: frontend/appflowy_flutter/lib/plugins/database_view/widgets/row/row_property.dart [hunk 3]
```diff
@@ -140,7 +143,28 @@ class _PropertyCellState extends State<_PropertyCell> {
...
+            child: _isFieldHover
+                ? BlockActionButton(
+                    onTap: () => _popoverController.show(),
+                    svg: FlowySvgs.drag_element_s,
+                    richMessage: TextSpan(
+                      text:
+                          LocaleKeys.grid_rowPage_fieldDragEelementTooltip.tr(),
+                    ),
+                  )
+                : null,
```
Comment (richardshiue) [inline frontend/appflowy_flutter/lib/plugins/database_view/widgets/row/row_property.dart:165]: You should be checking _isFieldHover before AppFlowyPopover

**Review by richardshiue — APPROVED** [review]
LGTM

**Comment by ekakshjanweja** [comment]
@richardshiue Shall we merge this then?

**Comment by richardshiue** [comment]
Merging despite integration test fails. failed test on ubuntu succeeded on windows and vice versa.

**Comment by ekakshjanweja** [comment]
@richardshiue @Xazin @LucasXu0 Thanks for helping me out throughout the process 🙏. I will be now looking forward to more issues.

**Comment by richardshiue** [comment]
Thanks for your contribution! Looking forward to your next one!

**Pull request merged**